main/src/main.ts

```diff
@@ -183,6 +182,16 @@ export function createWindow(commandLineArguments, workingDirectory) {
     windows.push(mainWindow)
 }
 
+app.on("open-file", (event, filePath) => {
+    event.preventDefault()
+    Log.info(`filePath to open: ${filePath}`) // tslint:disable-line
+    if (mainWindow) {
+        mainWindow.webContents.send("open-file", filePath)
+    } else if (process.platform.includes("darwin")) {
+        process.argv.push(filePath)
```

Comment:
In this case, where there is not an existing `mainWindow` instance, do we need to create a window? Or is some other codepath hit that we expect the window to be created?

I tested this out, and in the case where I had closed Oni (but it was still in the dock - so there is a `main` process but no `mainWindow`/renderer process active), and I tried to open a file with Oni, it never opened - makes me think perhaps we need to create a `BrowserWindow` here?

Reply:
@bryphe did you test this with the prev. pr merged #1241, I sort of created these three PRs with each other in mind (basically just sections of things ive tried in my local build) which made/makes it a tiny bit confusing but the logic there is that if you have the pr with the hide functionality then on clicking the icon the `activate` event is dispatched which should trigger a create window if none exists in which case a create window is in progress and so I push the missing file path with that in mind, they might not work as smoothly together anymore

Reply:
They're all a little inter-related unfortunately #1245  #1241  and #1246, I was trying to solve each locally and once I did I split the changes into what I hoped where separate PR's rather than one amorphous giant one, apologies didn't forsee their interdependency ☹️

Reply:
Actually nevermind all that it does cause the same issue for me locally 😆 does need a create window there (not so interdependent just a bug

Reply:
Haha cool, thanks for checking it out 😄